src/modules/dashboard/components/Dialogs/GnosisControlSafeDialog/TransactionTypesSection/ContractInteractionSection.tsx

```diff
@@ -47,7 +74,104 @@ const renderAvatar = (address: Address, item: AnyUser) => (
 const ContractInteractionSection = ({
   disabledInput,
   transactionFormIndex,
+  values,
+  setFieldValue,
+  selectedContractMethods = {},
+  handleSelectedContractMethods,
+  safes,
 }: Props) => {
+  const [formattedMethodOptions, setFormattedMethodOptions] = useState<
+    SelectOption[]
+  >([]);
+
+  const transactionValues = values.transactions[transactionFormIndex];
+  const contractAddress = transactionValues.contract?.profile?.walletAddress;
+  const onContractABIChange = useCallback(
+    (abi: string) => {
+      if (abi !== transactionValues.abi) {
+        const updatedSelectedContractMethods = omit(
+          selectedContractMethods,
+          transactionFormIndex,
+        );
+
+        setFieldValue(`transactions.${transactionFormIndex}.abi`, abi);
+        setFieldValue(
+          `transactions.${transactionFormIndex}.contractFunction`,
+          '',
+        );
+        handleSelectedContractMethods(updatedSelectedContractMethods);
+      }
+    },
+    [
+      transactionFormIndex,
+      setFieldValue,
+      transactionValues.abi,
+      selectedContractMethods,
+      handleSelectedContractMethods,
+    ],
+  );
+  const selectedSafe = safes.find((safe) => safe.address === values.safe);
+
+  const usefulMethods: AbiItemExtended[] = useMemo(
+    () =>
+      getContractUsefulMethods(
+        contractAddress,
+        transactionValues.abi,
+        Number(selectedSafe?.chain) || GNOSIS_NETWORK.chainId,
```

Comment:
In the mock data, the chain is the name, not chainId, so this will also default to Gnosis.